analyzers/src/SonarAnalyzer.CSharp/Rules/TypeNamesShouldNotMatchNamespaces.cs

```diff
@@ -39,16 +40,15 @@ public sealed class TypeNamesShouldNotMatchNamespaces : SonarDiagnosticAnalyzer
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         // Based on https://msdn.microsoft.com/en-us/library/gg145045%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396
-        private static readonly ISet<string> FrameworkNamespaces =
-            new SortedSet<string>(StringComparer.InvariantCultureIgnoreCase)
+        private static readonly ISet<string> FrameworkNamespaces = new HashSet<string>(StringComparer.InvariantCultureIgnoreCase)
             {
                 "Accessibility", "Activities", "AddIn", "Build", "CodeDom", "Collections",
                 "Componentmodel", "Configuration", "CSharp", "CustomMarshalers", "Data",
                 "Dataflow", "Deployment", "Device", "Diagnostics", "DirectoryServices",
                 "Drawing", "Dynamic", "EnterpriseServices", "Globalization", "IdentityModel",
                 "InteropServices", "IO", "JScript", "Linq", "Location", "Management", "Media",
                 "Messaging", "Microsoft", "Net", "Numerics", "Printing", "Reflection", "Resources",
-                "Runtime", "security", "server", "servicemodel", "serviceprocess", "speech",
+                "Runtime", "Security", "Server", "Servicemodel", "Serviceprocess", "Speech",
```

Comment:
For more consistency
```suggestion
                "Runtime", "Security", "Server", "ServiceModel", "ServiceProcess", "Speech",
```